### ISSUE TITLE
postponed mails get 'lost' with specific configurations

### ISSUE THREAD

**Issue by freshprince** [neomutt/neomutt#1011]
* *Expected behavior*
In my config I'd like to have my postponed messages stored in the corresponding IMAP-directories that I set via `folder_hook` (I set `$postponed=+Drafts` and  `$folder=imaps://user@host.com/` via folder_hook)

* *Actual behavior*
My postponed mails end up in my CWD with a garbled filename containing some random bytes.

* *Steps to reproduce*
Here's a minimal `muttrc` and the steps to reproduce:
```
# content of muttrc:
folder-hook . set postponed=+postponed
folder-hook .*abcd set postponed=+nasty

set folder="mbox"
unset spoolfile

# steps to reproduce:
#
# $ neomutt -F muttrc
# say no to 'mbox does not exist. Create it? ([yes]/no):' "n"
# create mail "m"
# To: "test"
# Subject: "test"
# exit editor ":wq"
# abort mail "q"
# say yes to 'Postpone this message? ([yes]/no):' "y"
# quit neomutt "q"
# $ ls
```

* *Used program versions*
neomutt-20171215
* *Operating System and its version*
voidlinux glibc x86_64 with linux 4.14.5

**Comment by flatcap** [comment]
Notes to self:
This one's a bit tricky
1. `ci_send_message()` passes `$postponed` to
1. `mutt_write_fcc()` which runs a `folder-hook`
1. which sets `$postponed` deleting the original version of the string

This bug probably happens in the same way when sending email, if a `folder-hook` for "Sent" sets `$record`.

Todo:
1. Move the two `folder-hook`s out of `mutt_write_fcc()` -- for each of the callers
1. Check if we should be running the `account-hook` too

**Comment by Dabsen** [comment]
A different, but presumably related, problem started affecting me recently: postponement often does not work at all any longer, i.e., when mutt asks, 'Postpone this message? ([yes]/no):', typing 'y' has no effect. The prompt stays in place and nothing happens until I type 'n'. It does not always happen; it might depend on the underlying folder but I haven't had time to test the behaviour systematically under various conditions and with a minimal muttrc.

So a quick question: would this be an obvious consequence of the same bug? If not, I'll look into it and open a second issue.

**Comment by cache-return** [comment]
I have this problem with no folder hooks and "postponed = ~/.mutt/oflineimap/account/INBOX/Drafts" and/or with folder set and "postponed = +INBOX/Drafts".
Hitting "P" creates a file with a name like ''$'\360\364\247\224''CV' in my ~ directory, and recall has no effect.

Unknown if related, but "P" doesn't work at all in the compose screen unless I scroll down to the second or subsequent email attachment.

**Comment by freshprince** [comment]
This seems to be fixed in neomutt-20180223.  Thank you!
